lib/redis.js

```diff
@@ -32,63 +33,77 @@ module.exports = class Redis {
     ];
   }
 
-  constructor(host, ttl) {
-    this._host = host;
-    this._ttl = ttl || 7200;
+  constructor(redisUrl, redisTTL = 7200, maxRetries = 5) {
+    this.disconnected = true;
+    this.url = redisUrl;
+    this.ttl = redisTTL;
+    this.conn = Redis.buildConn(redisUrl, maxRetries);
+    this.conn.on('connect', () => this.disconnected = false);
+    this.conn.on('error', err => {
+      logger.error('Redis error:', err);
+      this.disconnected = true;
+    });
   }
 
-  hostName() {
-    if (this._host) {
-      return 'redis://' + this._host.replace(/^.*\/\//, '').replace(/:.*$/, '');
+  static buildConn(redisUrl, maxRetries) {
+    const opts = {
+      lazyConnect: true,
+      maxRetriesPerRequest: maxRetries,
+      retryStrategy: times => {
+        if (times <= maxRetries) {
+          return Math.min(times * 50, 2000);
+        } else {
+          this.disconnected = true;
+          return null;
+        }
+      }
+    };
+
+    // optionally run in cluster mode
+    if (redisUrl.match(/^cluster:\/\/.+:[0-9]+$/)) {
```

Comment:
Is this for the dt-router redis cluster?

Reply:
Yeah, for some reason `ioredis` isn't great at detecting if the url it hits is clustered-or-nonclustered redis.  So you have to tell it upfront what to expect.  `cluster://1.2.3.4` is just a scheme i made up to do that!